src/GraphQL/Definitions.hs

```diff
@@ -60,17 +60,13 @@ instance forall a as. (HasFieldDefinition a, HasFieldDefinitions as) => HasField
 instance HasFieldDefinitions '[] where
   getFieldDefinitions = []
 
--- Convert a type-level list of KnownSymbol into a value-level
--- list. This is used for Enums.
-class GetSymbolList a where
-  getSymbolList :: [Text]
-
-instance forall a as. (KnownSymbol a, GetSymbolList as) => GetSymbolList (a:as) where
-  getSymbolList = (toS (symbolVal (Proxy :: Proxy a))):(getSymbolList @as)
-
-instance GetSymbolList '[] where
-  getSymbolList = []
-
+-- | For each enum type we need 1) a list of all possible values 2) a
+-- way to serialise and 3) deserialise.
+class GraphQLEnum a where
+  enumValues :: [Text] -- todo maybe use GValue.Value instead of Text?
+  enumFromValue :: GValue.Value -> Either Text a
+  enumToValue :: a -> GValue.Value
```

Comment:
I'm not a huge fan of these names, but can't articulate why and think of anything better at the moment. As mentioned in another comment, we should do a naming / vocabulary review before releasing.